### PR TITLE
[EA Forum only] donation election voting homepage banner

### PR DESCRIPTION
**Pull request by s-cheng**
This PR adds a version of the EA Forum homepage banner that will appear to users who are eligible to vote in the donation election (and have not already voted). Users who don't qualify see the current version of the banner. [Figma here](https://www.figma.com/file/3WOTZHH10ZBXSZPWw9omM3/Giving-Season?node-id=0%3A1&mode=dev)

The figma only had one screen size (the largest), so I made up the rest. Generally things look similar and just get smaller as the screen width decreases. I recommend looking at it at various screen sizes. It disappears under 300px (like the current banner does).

I didn't time-gate this version of the banner, so that we can test it more easily, and I thought we might prefer to manually control when it's up. If we time-gate the voting portal we should prob add that logic here too.

Screen Shot 2023-11-30 at 12 58 28 AM

Screen Shot 2023-11-30 at 12 59 26 AM

Screen Shot 2023-11-30 at 12 59 48 AM

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1206062738058631) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/lib/eaGivingSeason.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ export const votingThankYouImageId = "voting_portal_hero";
 
 const votingAccountCreationCutoff = new Date("2023/10/23");
 
-const userCanVoteInDonationElection = (
+export const userCanVoteInDonationElection = (
```
Comment (Will-Howard) [inline packages/lesswrong/lib/eaGivingSeason.ts:26]: Ah I hadn't see this existed, I was doing a similar thing in https://github.com/ForumMagnum/ForumMagnum/pull/8340 . I'll switch to using this in that brach

**Comment by Will-Howard** [comment]
>If we time-gate the voting portal we should prob add that logic here too.

I'm not going to time-gate when the voting portal opens (but I will for when it closes), so it seems fine to just wait to deploy this

**Review by Will-Howard — APPROVED** [review]
LGTM!

**Comment by agnesstenlund** [comment]
- [ ] oh also the different weeks should still be clickable (using the same links as before)

File: packages/lesswrong/components/ea-forum/giving-portal/GivingSeasonBanner.tsx [hunk 2]
```diff
@@ -180,24 +292,31 @@ const BannerDate: FC<{
...
         <div className={classes.givingSeasonGradient} />
-        <div className={classes.overview}>
+        <div className={classNames(classes.overview, {[classes.votingOverview]: advertiseVoting})}>
           <Typography
             variant="display1"
-            className={classes.heading}
+            className={classNames(classes.heading, {[classes.votingHeading]: advertiseVoting})}
           >
-            Giving season 2023
+            {advertiseVoting ? 'Vote in the Donation Election 2023' : 'Giving season 2023'}
```
Comment (agnesstenlund) [inline packages/lesswrong/components/ea-forum/giving-portal/GivingSeasonBanner.tsx:317]:
```suggestion
            {advertiseVoting ? 'Where should we donate?' : 'Giving season 2023'}
```

**Comment by agnesstenlund** [comment]
Looks great to me! I changed the title text, should be ready to go 🚢

**Pull request merged**